Added CIDR block set to datasource VPC peering

AlbertoSH opened the pull request:
Current data source `aws_vpc_peering_connection` only provides [`cidr_block`](https://www.terraform.io/docs/providers/aws/d/vpc_peering_connection.html#cidr_block) and [`peer_cidr_block`](https://www.terraform.io/docs/providers/aws/d/vpc_peering_connection.html#peer_cidr_block). This works fine for most scenarios in which you have two VPCs with just their primary CIDR, but when you attach secondary CIDRs, that info is missing

This PR adds both a `cidr_block_set` and a `cidr_block_set` attributes to the data source exposing those values


<!--- See what makes a good Pull Request at : https://github.com/terraform-providers/terraform-provider-aws/blob/master/.github/CONTRIBUTING.md#pull-requests --->

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

<!--- Thank you for keeping this note for the community --->

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):
<!--
If change is not user facing, just write "NONE" in the release-note block below.
-->

```release-note
- Added `cidr_block_set` & `peer_cidr_block_set` to `aws_vpc_peering_connection` data source.
```

Output from acceptance testing:

<!--
Replace TestAccXXX with a pattern that matches the tests affected by this PR.

For more information on the `-run` flag, see the `go test` documentation at https://tip.golang.org/cmd/go/#hdr-Testing_flags.
-->
```
$ make testacc TEST=./aws TESTARGS='-run=TestAccDataSourceAwsVpcPeeringConnection'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccDataSourceAwsVpcPeeringConnection -timeout 120m
=== RUN   TestAccDataSourceAwsVpcPeeringConnection_basic
=== PAUSE TestAccDataSourceAwsVpcPeeringConnection_basic
=== RUN   TestAccDataSourceAwsVpcPeeringConnection_cidBlockSets
=== PAUSE TestAccDataSourceAwsVpcPeeringConnection_cidBlockSets
=== CONT  TestAccDataSourceAwsVpcPeeringConnection_basic
=== CONT  TestAccDataSourceAwsVpcPeeringConnection_cidBlockSets
--- PASS: TestAccDataSourceAwsVpcPeeringConnection_basic (69.24s)
--- PASS: TestAccDataSourceAwsVpcPeeringConnection_cidBlockSets (95.30s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	97.233s
```

AlbertoSH commented:
If you consider so, I can move the included checks into a new test so the existing one remains as simple as it was before

My Golang capabilities are quite limited, I did my best 🙃

ewbankkit commented:
@AlbertoSH Thanks for submitting this. The Go looks perfect 😄.
Yes, we prefer additional test(s) for new attributes to ensure that we introduce no regressions.
Also, please update the relevant documentation: `website/docs/d/vpc.html.markdown`.

AlbertoSH commented:
Hi @ewbankkit 

I moved the checks to a new test & updated the docs. I left the existing test as it was and both tests are green (I updated the PR description with this new output)

```
$ make testacc TEST=./aws TESTARGS='-run=TestAccDataSourceAwsVpcPeeringConnection'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccDataSourceAwsVpcPeeringConnection -timeout 120m
=== RUN   TestAccDataSourceAwsVpcPeeringConnection_basic
=== PAUSE TestAccDataSourceAwsVpcPeeringConnection_basic
=== RUN   TestAccDataSourceAwsVpcPeeringConnection_cidBlockSets
=== PAUSE TestAccDataSourceAwsVpcPeeringConnection_cidBlockSets
=== CONT  TestAccDataSourceAwsVpcPeeringConnection_basic
=== CONT  TestAccDataSourceAwsVpcPeeringConnection_cidBlockSets
--- PASS: TestAccDataSourceAwsVpcPeeringConnection_basic (69.24s)
--- PASS: TestAccDataSourceAwsVpcPeeringConnection_cidBlockSets (95.30s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	97.233s
```

ewbankkit commented:
Verified acceptance tests:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccDataSourceAwsVpcPeeringConnection_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -count 1 -parallel 20 -run=TestAccDataSourceAwsVpcPeeringConnection_ -timeout 120m
=== RUN   TestAccDataSourceAwsVpcPeeringConnection_basic
=== PAUSE TestAccDataSourceAwsVpcPeeringConnection_basic
=== RUN   TestAccDataSourceAwsVpcPeeringConnection_cidBlockSets
=== PAUSE TestAccDataSourceAwsVpcPeeringConnection_cidBlockSets
=== CONT  TestAccDataSourceAwsVpcPeeringConnection_basic
=== CONT  TestAccDataSourceAwsVpcPeeringConnection_cidBlockSets
--- PASS: TestAccDataSourceAwsVpcPeeringConnection_basic (41.27s)
--- PASS: TestAccDataSourceAwsVpcPeeringConnection_cidBlockSets (64.32s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	64.353s
```

ewbankkit commented on aws/data_source_aws_vpc_peering_connection_test.go:
```diff
@@ -41,6 +41,25 @@ func TestAccDataSourceAwsVpcPeeringConnection_basic(t *testing.T) {
 	})
 }
 
+func TestAccDataSourceAwsVpcPeeringConnection_cidBlockSets(t *testing.T) {
```
Nit: Rename to `TestAccDataSourceAwsVpcPeeringConnection_cidrBlockSets()`.

AlbertoSH replied:
Good catch!
Renamed

AlbertoSH commented:
Hi @ewbankkit 
Is there anything else needed to merge the PR? We (my team) would like to have this feature available in a regular provider release 🙂

Amzani commented:
Hey guys,

Any updates on this.
We really need it.

artem-timchenko commented:
Hi guys, 
Any plans to merge this? This feature will be really helpful

Amzani commented:
@ewbankkit when do you guys plan to merge this ?

ewbankkit commented:
@AlbertoSH This is looking good. Can you please convert the new acceptance tests config to Terraform 0.12 syntax?

For example

```hcl
"${aws_vpc.foo.id}"
```

to

```hcl
aws_vpc.foo.id
```

AlbertoSH commented:
@ewbankkit There you go!
:)

DrFaust92 reviewed (approved):
LGTM

```
--- PASS: TestAccDataSourceAwsVpcPeeringConnection_basic (71.73s)
--- PASS: TestAccDataSourceAwsVpcPeeringConnection_cidrBlockSets (98.37s)
```

bflad commented on aws/data_source_aws_vpc_peering_connection.go:
```diff
@@ -40,6 +40,19 @@ func dataSourceAwsVpcPeeringConnection() *schema.Resource {
 			},
+			"cidr_block_set": {
+				Type:     schema.TypeList,
+				Optional: true,
```
Since the attribute is not used for configuring the data source, `Optional` should be removed. 👍 

```suggestion
```

bflad commented on aws/data_source_aws_vpc_peering_connection.go:
```diff
@@ -60,6 +73,19 @@ func dataSourceAwsVpcPeeringConnection() *schema.Resource {
 			},
+			"peer_cidr_block_set": {
+				Type:     schema.TypeList,
+				Optional: true,
```
Similarly:

```suggestion
```

bflad commented on website/docs/d/vpc_peering_connection.html.markdown:
```diff
@@ -82,6 +82,10 @@ All of the argument attributes except `filter` are also exported as result attri
 * `requester` - A configuration block that describes [VPC Peering Connection]
 (https://docs.aws.amazon.com/vpc/latest/peering/what-is-vpc-peering.html) options set for the requester VPC.
 
+* `cidr_block_set` - (Optional) The list of all CIDR blocks of the requester VPC of the specific VPC Peering Connection to retrieve. 
```
```suggestion
* `cidr_block_set` - List of all CIDR blocks of the requester VPC. 
```

bflad commented on website/docs/d/vpc_peering_connection.html.markdown:
```diff
@@ -82,6 +82,10 @@ All of the argument attributes except `filter` are also exported as result attri
 
+* `cidr_block_set` - (Optional) The list of all CIDR blocks of the requester VPC of the specific VPC Peering Connection to retrieve. 
+
+* `peer_cidr_block_set` - (Optional) The list of all CIDR blocks of the accepter VPC of the specific VPC Peering Connection to retrieve.
```
```suggestion
* `peer_cidr_block_set` - List of all CIDR blocks of the accepter VPC.
```

bflad reviewed (commented):
Hi @AlbertoSH 👋  Thank you for submitting this, overall this was looking good, just some minor changes we will complete on merge. 👍
